src/Aspire.Dashboard/Model/ResourceOutgoingPeerResolver.cs

```diff
@@ -14,10 +14,15 @@ public sealed class ResourceOutgoingPeerResolver : IOutgoingPeerResolver, IAsync
     private readonly CancellationTokenSource _watchContainersTokenSource = new();
     private readonly List<ModelSubscription> _subscriptions = [];
     private readonly object _lock = new();
-    private readonly Task _watchTask;
+    private readonly Task? _watchTask;
```

Comment:
You could set this to `Task.CompletedTask` if not enabled.

Reply:
True. `null` is fine for this though and avoids an await in the dispose. The nullability highlights that this value has two modalities.